### ISSUE TITLE
Do not crash JavaScript on PhpDebugBar not defined

### ISSUE THREAD

**Issue by brucek2** [barryvdh/laravel-debugbar#454]
I am separating this from the on-the-surface similar #418 because I believe a) the circumstance which triggers mine is different (I do not have the jQuery error first) and b) the fix requested is different.

The DebugBar functions fine on most of my application. However there is one section in which a home / landing page performs redirects based on cookie state. In this section, the debugbar sometimes loads (usually the first time in a session), and often not (such as on refresh.) It seems that in this case somehow the portion of the injected JS that requires "new PhpDebugBar.DebugBar()" to succeed is injected, but another earlier portion which actually defines the PhpDebugBar class is not.

While it would be nice to know and fix the underlying cause, I would settle for what I assume is a much easier fix to surround that second part of the code with a conditional that checks if the class has been defined so that when this happens, the rest of the app can proceed. (Or alternatively, to inject the two parts together so they will either both load or both not load.)

**Comment by Talean-dev** [comment]
similar problem

**Comment by brucek2** [comment]
I figured out the root cause for my issue: custom middleware in my application was blocking the requests for _debugbar/assets/javascript and _debugbar/assets/stylesheets. I added an exception for them to work around the issue. I did not think to do this previously because normally I would not expect requests for js/css assets to pass through the Laravel apps, vs. be served directly by the web server.

Although this fixes my immediate symptom, I still feel it would make sense to check for the presence of PhpDebugBar in JavaScript before calling it and potentially generating a crash.
